### ISSUE TITLE
<b-table> Certain ARIA roles must contain particular children

### ISSUE THREAD

**Issue by briandhill** [bootstrap-vue/bootstrap-vue#3547]
### Describe the bug

When using `<b-table>` with data and fields props set to an empty array, the table isn't passing the accessibility requirement that [Certain ARIA roles must contain particular children](https://dequeuniversity.com/rules/axe/3.2/aria-required-children?application=AxeChrome). Reason both data and fields arrays are empty is because those are determined dynamically from the server response.

### Steps to reproduce the bug

Setup a `<b-table :data="[]" :fields="[]" />`
Run axe devtools to analyze.

### Expected behavior

If data is empty and fields is empty or isn't set, then there should be some kind of default behavior for the table to pass accessibility if the server fails to load the data.

### Versions

**Libraries:**

- BootstrapVue: 2.0.0-rc.15
- Bootstrap: 4.3.1.
- Vue: 2.6.10

**Environment:**

- Device: hp
- OS: Windows 10
- Browser: Chrome
- Version: Latest

**Comment by tmorehouse** [comment]
Is this when using the `empty` slot, or jsut with the two empty arrays (items/fields)?

**Comment by tmorehouse** [comment]
Also, which elements/roles are generating the issue?

**Comment by briandhill** [comment]
So the actual version we're using is using the empty slot as well, but I notice the `<thead>` and `<tbody>` being generated without it too. From what I noticed, it's a `<tr>` being empty inside the `<thead>` and `<tbody>`.

**Comment by tmorehouse** [comment]
We may need to just add a conditional render for b-table, and render a placeholder comment if both `fields` and `items` are empty, and no empty slot present (there needs to be at least once header rendered for empty slot to work currently)

**Comment by briandhill** [comment]
Even when empty slot is used, the accessibility requirement is invalid for the `<thead>`.

**Comment by briandhill** [comment]
Could you not render the '<thead>' unless there is data or fields set?

**Comment by tmorehouse** [comment]
Version 2.0.0-rc.25 has been released.

**Comment by briandhill** [comment]
Thanks man!
